OpenCL_C.txt

```diff
@@ -2056,73 +2056,46 @@ types.
 
 [open,refpage='addressSpaceQualifiers',desc='Address Space Qualifiers',type='freeform',spec='clang',anchor='address-space-qualifiers',xrefs='constant genericAddressSpace global local private']
 --
-
-OpenCL implements the following disjoint named address spaces: `+__global+`,
-`+__local+`, `+__constant+` and `+__private+`.
+OpenCL has hierarchical memory architecture represented by address spaces
+defined in <<embedded-c-spec, section 5 of Embedded C Specification>>. It
+extends C syntax to allow an address space name as a valid type qualifier
+(<<embedded-c-spec, section 5.1.2 of Embedded C>>).
+OpenCL implements the following disjoint named address spaces with the spelling:
+`+__global+`, `+__local+`, `+__constant+` and `+__private+`.
 The address space qualifier may be used in variable declarations to specify
-the region of memory that is used to allocate the object.
-The C syntax for type qualifiers is extended in OpenCL to include an address
-space name as a valid type qualifier.
-If the type of an object is qualified by an address space name, the object
-is allocated in the specified address space name.
-
-The address space names without the `+__+` prefix, i.e. `global`, `local`,
-`constant` and `private`, may be substituted for the corresponding address
-space names with the `+__+` prefix.
-
-The address space name for arguments to a function in a program, or local
-variables of a function is `+__private+`.
-All function arguments shall be in the `+__private+` address space.
+the region where objects are to be allocated. If the type of an
+object is qualified by an address space name, the object is allocated in the
+specified address space. Similarly pointer types can be qualified by an
+address space representing a memory segment pointed to.
```

Comment:
Changed everywhere apart from:

> Private address space is a memory segment that can only be accessed by one work item.

Here I use "memory segment" as a generic concept representing physical HW rather than a language feature in OpenCL C. Do you think it is reasonable? Otherwise, I will try to change the wording.